ansible_base/lib/utils/create_system_user.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from typing import Optional, Tuple
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.translation import gettext as _
+
+from ansible_base.lib.utils.settings import get_setting
+
+logger = logging.getLogger('ansible_base.lib.utils.create_system_user')
+
+"""
+These functions are in its own file because it is loaded during migrations so it has no access to models.
+"""
+
+
+def create_system_user(user_model=None) -> object:  # Note: We can't load models here so we can typecast to anything better than an object
+    #
+    # Creating the system user is a little tricky because we need to reference ourselves
+    #
+    if not user_model:
+        logger.error("You must pass a user_model to create_system_user")
+        return None
+
+    # First create a User object for the system user
+    system_user, created = user_model.objects.get_or_create(username=get_system_username()[0])
+
+    if created:
+        logger.info(f"Created system user {system_user.username}")
+    else:
+        logger.debug("System user is already created")
+        return system_user
+
+    system_user.is_active = False
```

Comment:
You could probably move this, and maybe only this, into `defaults` for the `get_or_create`.

Reply:
Yea, I thought about doing this but since we have to call save either way I didn't think it would make a difference.